h5netcdf/core.py

```diff
@@ -125,28 +132,62 @@ def shuffle(self):
 
 NOT_A_VARIABLE = b'This is a netCDF dimension but not a netCDF variable.'
 
+class lazy_objects():
```

Comment:
This should inherit from `collections.Mapping` to ensure it satisfies the mapping interface